README.md

```diff
@@ -2,4 +2,63 @@
 
 **THIS IS DRAFT AF. DON'T READ IT IF YOU DON'T LIKE CONFUSION.**
 
-## TOC
\ No newline at end of file
+## TOC
+
+* [Data Distribution, Not State Management](#data-distribution-not-state-management)
+
+## Data Distribution, Not State Management
+
+Context makes it possible to distribute data to every component in a component tree.
+
+It's used to distribute data, not manage state.
+That said, it provides the mechanism needed to state and updater functions managed by state containers.
+
+Here's an example of a stateful container that uses Context to distribute local `state` and an `update` function.
+
+```jsx
+let StateContext = React.createContext();
+
+class StateProvider extends React.Component {
+  static defaultProps = {
+    initialState: {}
+  };
+
+  state = { ...this.props.initialState };
+
+  update = (...args) => this.setState.apply(this, args);
+
+  render() {
+    return (
+      <StateContext.Provider
+        value={{
+          state: this.state,
+          update: this.update
```

Comment:
This is not super great because it will update all consumers on every re-render. 
https://reactjs.org/docs/context.html#caveats

Reply:
i'd love to be able to answer the question "how do i update state through context simple?". The intended audience being UI developers using context for localized state (.e.g, <MediaPlayer />).

in this case, would the recommendation be to "not use context" and use use a more explit pattern?

Reply:
I'd like to illustrate the problem and then offer options for folks that actually bump into it.

Is there something that sits between this (updates every Consumer, every update) and using a side-loading state management library?

Reply:
I just mean that you need to put the whole `{ state: this.state, update: this.update }` in the state. So that you can pass `value={this.state}` and not recreate it every time. This is not a philosophical concern, just a practical one.

Reply:
Something like this?

```js
class StateProvider extends React.Component {
  static defaultProps = {
    initialState: {}
  };

  update = (updater, done) => {
    this.setState(prevState => ({
     current: typeof updater === 'function' ?
      updater(prevState.current) :
      updater
    }));
  };

  state = {
    current: this.props.initialState,
    update: this.update
  };

  render() {
    return (
      <StateContext.Provider
        value={this.state}
        {...this.props}
      />
    );
  }
}
```

Reply:
aaaaah. I see what you’re saying now. Thanks